dds/idl/langmap_generator.cpp

```diff
@@ -1421,6 +1421,25 @@ struct Cxx11Generator : GeneratorBase
   bool scoped_enum() { return true; }
   std::string enum_base() { return " : uint32_t"; }
 
+  void gen_union_pragma_pre()
+  {
+    // Older versions of gcc will complain because it appears that a primitive
+    // default constructor is not called for anonymous unions.
+    be_global->lang_header_ <<
+      "#ifdef __GNUC__\n"
```

Comment:
Maybe generate a gcc version check? That way you know when to drop this workaround?

Reply:
Maybe.  I'd have to test which versions report the warning.

Reply:
Instead of suppressing the warning, you can also use default/zero initialization, see TAOX11. See https://github.com/RemedyIT/idl2cpp11/blob/39ca4d118e9bf51e8f25cad906926472b857e6a2/examples/union/testC.h#L313 for some example code, I have the PR https://github.com/RemedyIT/taox11/pull/239 to reduce the amount of ifdef checks, only for the member that has zero initialization we need the ifdef

Reply:
I think this situation is slightly different.  This is an anonymous union relying on the default constructor.  So, when there is a string in the union, it calls that constructor.  But, when it does not contain a complex data type, it **should** call a constructor for one of the types.  I believe it actually does.  But, the gcc warning doesn't realize it?

Reply:
We had similar warnings related to unions and switched to use default which solved it for us. Maybe handcraft it and see if this also helps with opendds

Reply:
That's a good suggestion.  That may get done for v4.

Reply:
Also changing optimization level will probably help